src/Raven.Server/Web/Studio/DataDirectoryInfo.cs

```diff
@@ -0,0 +1,157 @@
+        private static readonly Logger Logger = LoggingSource.Instance.GetLogger<DataDirectoryInfo>("DataDirectoryInfo");
+
+        private readonly ServerStore _serverStore;
+        private readonly JsonOperationContext _context;
+
+        public DataDirectoryInfo(ServerStore serverStore, JsonOperationContext context)
+        {
+            _serverStore = serverStore;
+            _context = context;
+        }
+
+        public async Task<DataDirectoryResult> GetDatabaseDirectoryResult(
+            SingleNodeDataDirectoryResult currentNodeInfo, string path, string name)
+        {
+            var clusterTopology = _serverStore.GetClusterTopology();
+            var relevantNodes = clusterTopology.AllNodes.Keys;
+            var urlPath = $"admin/studio-tasks/full-data-directory?path={path}&name={name}";
+
+            var dataDirectoryResult = new DataDirectoryResult();
+            dataDirectoryResult.List.Add(currentNodeInfo);
+            
+            if (relevantNodes.Count > 1)
+            {
+                await UpdateNodesDirectoryResult(relevantNodes, clusterTopology, urlPath, dataDirectoryResult);
+            }
+
+            return dataDirectoryResult;
+        }
+
+        private async Task UpdateNodesDirectoryResult(
+            IEnumerable<string> nodes, ClusterTopology clusterTopology,
+            string urlPath, DataDirectoryResult dataDirectoryResult)
+        {
+            var httpClient = RequestExecutor.CreateHttpClient(_serverStore.Server.Certificate.Certificate, DocumentConventions.Default);
+            var tasks = new List<Task<SingleNodeDataDirectoryResult>>();
+
+            foreach (var nodeTag in nodes)
+            {
+                _serverStore.ServerShutdown.ThrowIfCancellationRequested();
+
+                if (nodeTag.Equals(_serverStore.NodeTag, StringComparison.OrdinalIgnoreCase))
+                    continue;
+
+                var serverUrl = clusterTopology.GetUrlFromTag(nodeTag);
+                if (serverUrl == null)
+                    continue;
+
+                tasks.Add(Task.Run(async () =>
+                {
+                    var url = $"{serverUrl}/{urlPath}";
+                    var singleNodeResult = await GetSingleNodeDataDirectoryInfo(url, httpClient);
+                    return singleNodeResult;
+                }, _serverStore.ServerShutdown));
+            }
+
+            await Task.WhenAll(tasks);
+
+            foreach (var task in tasks)
+            {
+                if (task.IsCompletedSuccessfully == false)
+                    continue;
+
+                var singleNodeResult = await task;
+                if (singleNodeResult == null)
+                    continue;
+
+                dataDirectoryResult.List.Add(singleNodeResult);
+            }
+        }
+
+        private async Task<SingleNodeDataDirectoryResult> GetSingleNodeDataDirectoryInfo(string url, HttpClient httpClient)
+        {
+            using (var cts = new CancellationTokenSource(7 * 1000))
```

Comment:
Why 7 seconds?

Reply:
No reason. Should it be lower?

Reply:
Better to take that as an argument and not hard code it

Reply:
done